src/niscope/examples/niscope_fetch_forever.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+
+import argparse
+import datetime
+import niscope
+import numpy as np
+import pprint
+import sys
+
+
+pp = pprint.PrettyPrinter(indent=4, width=80)
+
+
+# This example is based on the C FetchForever example
+def example(resource_name, channels, options, total_acquisition_time_in_seconds, voltage, sample_rate_in_hz, samples_per_fetch):
+    total_samples = int(total_acquisition_time_in_seconds * sample_rate_in_hz)
+    channel_list = channels.split(',')  # We need channels as a list
+
+    # 1. Creating numpy arrays
```

Comment:
* I suggest using array.array so you don't need to depend on numpy.
* Explain why we do it: As an optimization, so we can use `fetch_into` which allows us to re-use the buffers and save a reallocation per function call.

Reply:
* Fetch into doesn't support `array.array`
* Added overview